---
title: "naszegenomy_raport"
output: github_document
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(echo=FALSE)
suppressMessages(library(knitr))
```

```{r echo=FALSE}
outlier <- function (x) {
  IQR <- quantile(x)[[4]] - quantile(x)[[2]]
  low <- quantile(x)[[2]] - 1.5*IQR
  high <- quantile(x)[[4]] + 1.5*IQR
  out_val <- x[x < low | x > high]
  return(out_val)
}
```

```{r echo=FALSE}
suppressMessages(library(tidyverse))
suppressMessages(library(gridExtra))

flagstat_path = "output/multiqc_data/multiqc_samtools_flagstat.txt"
depth_path = "output/depth_concat.txt"

flagstat <- read.table(flagstat_path,header=T,sep='\t')
flagstat <- flagstat %>% 
    mutate(total_passed = total_passed/1e+06, properly.paired_passed = properly.paired_passed/1e+06, flagstat_total = flagstat_total/1e+06,
    mapped_passed = mapped_passed/1e+06, singletons_passed = singletons_passed/1e+06, duplicates_passed = duplicates_passed/1e+06)

```

## 1. Sequencing statistics (in millions)

```{r echo=FALSE}
x <- flagstat %>% select(flagstat_total,mapped_passed,properly.paired_passed) 
kable(summary(x))
```


```{r echo=FALSE}
p1 <- flagstat %>% ggplot(aes(x=flagstat_total)) + geom_boxplot() +
    xlab('Total reads (millions)') + theme_minimal() + xlim(c(600,1200)) + 
    theme(axis.text.y = element_blank())
p2 <- flagstat %>% ggplot(aes(x=mapped_passed)) + geom_boxplot() +
    xlab('Mapped (millions)') + theme_minimal()+ xlim(c(600,1200)) +
    theme(axis.text.y = element_blank())
p3 <- flagstat %>% ggplot(aes(x=properly.paired_passed)) + geom_boxplot() + 
    xlab('Properly paired (millions)') + theme_minimal()+ xlim(c(600,1200)) +
    theme(axis.text.y = element_blank())

grid.arrange(p1, p2, p3,ncol = 1)

```

## 2. Depth statistics

```{r echo=FALSE}
depth <- read.table(depth_path,header=T,sep='\t')

kable(depth %>% select(-sample) %>% summary())
```

### Average depth outliers 
```{r echo=FALSE}
kable(depth %>% select(sample,average_depth) %>% filter(average_depth %in% outlier(depth$average_depth)))
```

```{r echo=FALSE}
depth_intervals <- depth %>% select(-average_depth) %>% pivot_longer(cols = -sample,
                                      names_to = 'group', 
                                      values_to = 'depth'
                                      )
avg_depth_plot <- depth %>% ggplot(aes(x=average_depth)) + geom_boxplot() +
  theme_minimal() +
  theme(axis.text.y = element_blank())
avg_depth_plot
```


```{r echo=FALSE}
depth_percent <- depth_intervals %>% ggplot(aes(y= group, x=depth)) + 
    geom_boxplot() + 
    theme_minimal() +
    scale_y_discrete(labels=c("percentage_above_30" = "30x", "percentage_above_20" = "20x",
                                "percentage_above_10" = "10x")) +
    ylab('') +
    xlab('% coverage')
depth_percent
```

## 3. Per sample count (PSC)
```{r echo=FALSE}

psc <- read.table('output/total_psc.stats', sep='\t')
colnames(psc) <- c(
  'PSC','id', 'sample', 'nRefHom', 'nNonRefHom', 'nHets', 
                   'nTransitions', 'nTransversions', 'nIndels', 'average_depth',
                   'nSingletons', 'nHapRef', 'nHapAlt', 'nMissing'
)

kable(psc %>% select(-c(PSC,id,sample)) %>% summary())

p1 <- psc %>% ggplot(aes(x=nRefHom)) + geom_boxplot() +
    xlab('nRefHom') + theme_minimal() + theme(axis.text.y = element_blank()) + ggtitle('Number of reference homozygous')
p2 <- psc %>% ggplot(aes(x=nNonRefHom)) + geom_boxplot() +
    xlab('nNonRefHom') + theme_minimal() + theme(axis.text.y = element_blank()) + ggtitle('Number of non-reference homozygous')
p3 <- psc %>% ggplot(aes(x=nHets)) + geom_boxplot() +
    xlab('nHets') + theme_minimal() + theme(axis.text.y = element_blank()) + ggtitle('Number of heterozygous')

grid.arrange(p1, p2, p3,ncol = 1)
```
