---
title: "naszegenomy_raport"
output: github_document
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(echo=FALSE)
suppressMessages(library(knitr))
```

```{r echo=FALSE}
outlier <- function (x) {
  IQR <- quantile(x)[[4]] - quantile(x)[[2]]
  low <- quantile(x)[[2]] - 1.5*IQR
  high <- quantile(x)[[4]] + 1.5*IQR
  out_val <- x[x < low | x > high]
  return(out_val)
}
```

```{r echo=FALSE}
suppressMessages(library(tidyverse))
suppressMessages(library(gridExtra))

flagstat_path = "output/multiqc_data/multiqc_samtools_flagstat.txt"
depth_path = "output/depth_concat.txt"

flagstat <- read.table(flagstat_path,header=T,sep='\t')
flagstat <- flagstat %>% 
    mutate(total_passed = total_passed/1e+06, properly.paired_passed = properly.paired_passed/1e+06, flagstat_total = flagstat_total/1e+06,
    mapped_passed = mapped_passed/1e+06, singletons_passed = singletons_passed/1e+06, duplicates_passed = duplicates_passed/1e+06)

```

## 1. Sequencing statistics (in millions)

```{r echo=FALSE}
kable(flagstat %>% select(flagstat_total,mapped_passed,properly.paired_passed) %>%
  pivot_longer(cols = everything(), names_to = 'stat',values_to = 'value') %>%
  group_by(stat) %>% 
  summarise(min = round(min(value),2),median=round(median(value),2),mean=round(mean(value),2),
            max=round(max(value),2))
)
```


```{r echo=FALSE}
p1 <- flagstat %>% ggplot(aes(y=flagstat_total,x=1)) + geom_violin(fill='#48C095',col='#27384A') +
    geom_boxplot(width=0.1) +
    xlab('Total reads (millions)') + theme_classic() + ylim(c(600,1200)) + 
    theme(axis.text.x = element_blank())
p2 <- flagstat %>% ggplot(aes(y=mapped_passed,x=1)) + geom_violin(fill='#48C095',col='#27384A') +
    geom_boxplot(width=0.1) +
    xlab('Mapped (millions)') + theme_classic()+ ylim(c(600,1200)) +
    theme(axis.text.x = element_blank())
p3 <- flagstat %>% ggplot(aes(y=properly.paired_passed,x=1)) + geom_violin(fill='#48C095',col='#27384A') +
    geom_boxplot(width=0.1) + 
    xlab('Properly paired (millions)') + theme_classic()+ ylim(c(600,1200)) +
    theme(axis.text.x = element_blank())

grid.arrange(p1, p2, p3,nrow = 1)

```
Total reads outliers 
```{r echo=FALSE}
kable(flagstat %>% select(Sample,flagstat_total) %>% filter(flagstat_total %in% outlier(flagstat$flagstat_total)))
```



## 2. Depth statistics

```{r echo=FALSE}
depth <- read.table(depth_path,header=T,sep='\t')

kable(depth %>% 
  select(-sample) %>% 
  pivot_longer(cols = everything(), names_to = 'stat',values_to = 'value') %>%
  group_by(stat) %>% 
  summarise(min = round(min(value),2),median=round(median(value),2),mean=round(mean(value),2),
            max=round(max(value),2))
)
```

Average depth outliers 

```{r echo=FALSE}
kable(depth %>% select(sample,average_depth) %>% filter(average_depth %in% outlier(depth$average_depth)))
```

```{r echo=FALSE}
depth_intervals <- depth %>% select(-average_depth) %>% pivot_longer(cols = -sample,
                                      names_to = 'group', 
                                      values_to = 'depth'
                                      )
avg_depth_plot <- depth %>% ggplot(aes(y=average_depth,x=1)) + 
  geom_violin(fill='#48C095',col='#27384A') + 
  geom_boxplot(width=0.1) +
  theme_classic() +
  xlab('Average depth') + 
  ylab('') +
  theme(axis.text.x = element_blank())
avg_depth_plot
```


```{r echo=FALSE}
depth_percent <- depth_intervals %>% ggplot(aes(y= depth, x=group)) + 
    geom_boxplot() +
    theme_classic() +
    scale_y_discrete(labels=c("percentage_above_30" = "30x", "percentage_above_20" = "20x",
                                "percentage_above_10" = "10x")) +
    ylab('') +
    xlab('% coverage')
depth_percent
```

## 3. Per sample count (PSC)
```{r echo=FALSE}

psc <- read.table('output/total_psc.stats', sep='\t')
colnames(psc) <- c(
  'PSC','id', 'sample', 'nRefHom', 'nNonRefHom', 'nHets', 
                   'nTransitions', 'nTransversions', 'nIndels', 'average_depth',
                   'nSingletons', 'nHapRef', 'nHapAlt', 'nMissing'
)

kable(psc %>% 
  select(-c(PSC,id,sample,nHapAlt,nHapRef)) %>% 
  pivot_longer(cols = everything(), names_to = 'stat',values_to = 'value') %>%
  group_by(stat) %>% 
  summarise(min = round(min(value),2),median=round(median(value),2),mean=round(mean(value),2),
            max=round(max(value),2))
)

p1 <- psc %>% ggplot(aes(y=nRefHom,x=1)) + 
  geom_violin(fill='#48C095',col='#27384A') + 
  geom_boxplot(width=0.1) +
    ylab('Number of reference homozygous') + theme_classic() + theme(axis.text.x = element_blank()) +
  xlab('')
p2 <- psc %>% ggplot(aes(y=nNonRefHom,x=1)) + 
  geom_violin(fill='#48C095',col='#27384A') + 
  geom_boxplot(width=0.1) +
    ylab('Number of non-reference homozygous') + theme_classic() + theme(axis.text.x = element_blank())+
  xlab('')
p3 <- psc %>% ggplot(aes(y=nHets,x=1)) + 
  geom_violin(fill='#48C095',col='#27384A') + 
  geom_boxplot(width=0.1) +
    ylab('Number of heterozygous') + theme_classic() + theme(axis.text.x = element_blank())+
  xlab('')

grid.arrange(p1, p2, p3,nrow = 1)
```


## 3. AF
to do:
Summary: rare, medium, common variants, impacts

AF curves

```{r echo=FALSE}
library(png)
img1_path <- "raport_files/figure-gfm/afcurve.png"
img1 <- readPNG(img1_path, native = TRUE, info = TRUE)
include_graphics(img1_path)

```

## 4. ROHs

Summary

```{r echo=FALSE}
suppressMessages(library(data.table))
roh <- fread('input/roh_concat.txt', showProgress = T)
colnames(roh) <- c("Chromosome", "Start", 'End', 'Length', 'Number_of_markers', 'Quality','sample_id')
roh$sample_id <-gsub(".*/(.*)\\..*", "\\1", roh$sample_id)

kable(roh %>% select(Length,Number_of_markers,Quality) %>%
  pivot_longer(cols = everything(), names_to = 'stat',values_to = 'value') %>%
  group_by(stat) %>% 
  summarise(min = round(min(value),2),median=round(median(value),2),mean=round(mean(value),2),
            max=round(max(value),2))
)

```

ROHs quality histogram

```{r echo=FALSE}
roh  %>% ggplot(aes(x=1,y=Quality)) + 
    geom_violin(fill='#48C095', col='#27384A') + 
  geom_boxplot(width=0.1) +
     theme_classic() + theme(axis.text.x = element_blank()) +
  xlab('')
```

Number of ROHs with length in specific ranges

```{r echo=FALSE} 
test1 <- roh$Length > 5e+06 
test2 <- roh$Length < 25000

roh <- roh %>% 
    mutate(group = case_when(test1 ~ ">5Mb", 
                             (test2) ~ "<25Kb" , 
                             !test1 & !test2 ~ "25Kb-5Mb" 
    ))

roh  %>% ggplot(aes(x=1,y=Length)) + 
    geom_violin(fill='#48C095', col='#27384A') + geom_boxplot(width=0.1) +
     theme_classic() + theme(axis.text.x = element_blank()) + 
  facet_wrap(~group, ncol=3, scales = 'free_y') + xlab('')
```

to do: ROH fst

