---
title: "Alcohol metabolism"
subtitle: "Results on 943 unrelated individuals"
output:
  pdf_document: default
  github_document: default
---

```{r setup, echo=FALSE}
knitr::opts_chunk$set(echo=FALSE)
suppressMessages(library(knitr))
```




```{r echo=FALSE, message=F, warning=F}
suppressMessages(library(tidyverse))
suppressMessages(library(data.table))

genes <- fread('alko_genes_vep_input.tsv',header=T)
rsid <- fread('alko_rsid_vep_input.tsv',header=T)

rsid_comp <- rsid %>% 
  select(Existing_variation,Location,PL_AF,gnomADg_AF_NFE ) %>% distinct()

genes_comp <- genes %>% 
  select(Existing_variation,Location,PL_AF,gnomADg_AF_NFE ) %>% distinct()

alko_df <- rbind(rsid_comp,genes_comp)
alko_df <- alko_df %>% mutate(Association=ifelse(Existing_variation %in% rsid$Existing_variation,'confirmed',
                              'unconfirmed')
                                ) %>% filter(gnomADg_AF_NFE != '-')
alko_df$gnomADg_AF_NFE <- as.numeric(alko_df$gnomADg_AF_NFE)


alko_df$nfe_diff <- alko_df$PL_AF - alko_df$gnomADg_AF_NFE
```

Known variants related to alcohol metabolism

```{r, table_confirmedn, echo=FALSE}
kable(alko_df %>% filter(Location %in% rsid_comp$Location) %>%
        select(-Association) %>%
        mutate(PL_AF =  round(PL_AF,3),
               gnomADg_AF_NFE = round(gnomADg_AF_NFE,3),
               nfe_diff = round(nfe_diff,3)) %>% 
        rename(AF_NFE = gnomADg_AF_NFE) %>%
        arrange(nfe_diff) 
        )
```



\newpage

Top 10 and bottom 10 differences between AF in PL and gnomAD Non Finnish European

```{r, table_outlier, echo=FALSE}
af_outlier_gnomad <- alko_df %>% 
  top_n(-10, nfe_diff) %>% rbind( top_n(alko_df,10, nfe_diff)) %>% 
  arrange(nfe_diff)
kable(af_outlier_gnomad)
```




<!-- ```{r echo=FALSE, message=F} -->
<!-- af_comp %>%  -->
<!--   ggplot(aes(y=PL_AF,x=EUR_AF)) + geom_point(col='#27384A') +  -->
<!--   geom_smooth(method='lm',se = F,col='#48C095') + theme_classic() +  -->
<!--   xlab('AF in 1000 Genomes EUR population') + -->
<!--   ylab('AF in PL population') +  -->
<!--   ggtitle('Polish vs 1000 genomes European') + -->
<!--   theme(plot.title = element_text(hjust = 0.5)) -->
<!-- ``` -->
\newpage

1. Polish vs gnomAD Non-Finnish European for 29520 variants in genes:

```{r echo=FALSE}
kable(
  genes %>% group_by(SYMBOL) %>% summarise(variants = n()) 
)
```


```{r, Polish_gnomADNon-Finnish_scatter_plot, echo=FALSE, message=F}
group.colors <- c(confirmed = "#BC0020", unconfirmed ="#48C095")

alko_df %>% na.omit() %>%
  ggplot(aes(y=PL_AF,x=gnomADg_AF_NFE,col=Association)) + 
  geom_point(alpha=0.5) + 
  geom_smooth(method = 'lm',se=F,col='#27384A',formula = 'y~x') +
  theme_classic() + 
  xlab('gnomAD3g_AF_NFE') +
  ylab('AF in PL population') +
  scale_color_manual(values=group.colors)
```


2. Polish vs gnomAD Non-Finnish European for 23 variants with know association:

```{r known_variants_plot, echo=FALSE}
alko_df %>% filter(Association=='confirmed') %>% na.omit() %>%
  ggplot(aes(y=PL_AF,x=gnomADg_AF_NFE)) + 
  geom_point(col="#48C095") + 
  geom_smooth(method = 'lm',se=F,col='#27384A',formula = 'y~x') +
  theme_classic() + 
  xlab('gnomAD3g_AF_NFE') +
  ylab('AF in PL population') 

```


\newpage

3. Top 10 and bottom 10 between Polish and gnomAD Non Finnish European for all genes

```{r, 10_diff_barplot, echo=FALSE, message=F}
af_outlier_gnomad %>%
  ggplot(aes(x=nfe_diff*100,y = reorder(Existing_variation,-nfe_diff*100))) + 
  geom_bar(stat = 'identity',fill='#48C095') + theme_classic() +
  ylab('rsid ') + xlab('Difference between AF in Polish and European populations') +
  geom_text(aes(label=paste(round(nfe_diff*100),'%'))) +
  xlim(c(-100,100))
```

\newpage

4. Differences between Polish and gnomAD Non Finnish European for associated variants

```{r, known_diff_barplot, echo=FALSE, message=F}
alko_df%>% filter(Association=='confirmed') %>%
  ggplot(aes(x=nfe_diff*100,y = reorder(Existing_variation,-nfe_diff*100))) + 
  geom_bar(stat = 'identity',fill='#48C095') + theme_classic() +
  ylab('rsid ') + xlab('Difference between AF in Polish and European populations') +
  geom_text(aes(label=paste(round(nfe_diff*100,2),'%'))) +
  xlim(c(-10,10))
```
