---
title: "Variants in disease causing genes \newline Results for 943 unrelated individuals"
subtitle:
output: 
  github_document: default

---

```{r setup, echo=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo=FALSE, dev = 'jpeg')
suppressMessages(library(knitr))
suppressMessages(library(tidyverse))
suppressMessages(library(data.table))
library(grid)
suppressMessages(library(gridExtra))
```

## 1. Cummulative allele frequency


```{r af_hist,echo=FALSE}
af <- read.table('../input/multisample_20210519.dv.bcfnorm.filtered.ACgt0.AF_list.tsv')
colnames(af) <- c('AF','id', 'allele_frequency', 'SNP', 'number_of_transitions', 'number_of transversions', 'indel', 'repeat-consistent','repeat-inconsistent', 'not_applicable')

af$allele_frequency[1] <- 0.000530223

af_plot <- af %>% select(allele_frequency,SNP,indel) 
af_plot$SNP <- cumsum(af_plot$SNP)/1e+06
af_plot$indel <- cumsum(af_plot$indel)/1e+06
 

type.colors <- c(SNP = "#27384A", indel ="#48C095")

af_plot %>%
  pivot_longer(-allele_frequency,names_to = 'Variant class',
               values_to = 'Cummulative number of variants (millions)') %>%
  ggplot(aes(allele_frequency,`Cummulative number of variants (millions)`,
             col=`Variant class`)) +
  geom_line() +
  xlab('Allele frequency') +
  scale_color_manual(values = type.colors) +
  scale_y_continuous(breaks = seq(0,30,2)) +
  theme_classic()
  

```

<!-- ## 2. Allele count -->

<!-- fixing -->

<!-- ```{r ac_hist,echo=FALSE} -->
<!-- library(png) -->
<!-- img1_path <- "variants_af_files/figure-gfm/ac_hist.png" -->
<!-- img1 <- readPNG(img1_path, native = TRUE, info = TRUE) -->
<!-- include_graphics(img1_path) -->

<!-- ``` -->

```{r, echo=FALSE, warning=FALSE}
af_list <- c('PL_AF',  'gnomADg_AF_NFE','gnomADg_AF')
group.colors <- c(PL_AF = "#BC0020", AF = "#27384A", gnomAD_FIN_AF ="#48C095", gnomAD_AF_NFE = "#B6B6B6",EUR_AF='#5F5F5F')

```

## 3. ACMG

```{r ACMG, echo=FALSE, warning=FALSE}
acmg <- read.table('../input/diseases/acmg_ready.tsv',sep='\t',header=T)

comp_acmg <- acmg %>% 
    filter(PL_AC > 0) %>%
   select(Uploaded_variation,Existing_variation,Location,starts_with('ClinVar'), 
          SYMBOL,Gene, PL_AC, all_of(af_list)) %>%
  mutate(across(PL_AF:gnomADg_AF,as.character)) %>%
   mutate(across(PL_AF:gnomADg_AF,as.numeric))

comp_acmg$stars <- NA
comp_acmg$stars <- ifelse(comp_acmg$ClinVar_CLNREVSTAT == 'practice_guideline',
                             '4',comp_acmg$stars)
comp_acmg$stars <- ifelse(comp_acmg$ClinVar_CLNREVSTAT == 'reviewed_by_expert_panel',
                             '3',comp_acmg$stars)
comp_acmg$stars <- ifelse(
  comp_acmg$ClinVar_CLNREVSTAT == 'criteria_provided,_multiple_submitters,_no_conflicts',
                             '2',comp_acmg$stars)
comp_acmg$stars <- ifelse(comp_acmg$ClinVar_CLNREVSTAT == '_conflicting_interpretations',
                             '1',comp_acmg$stars)
comp_acmg$stars <- ifelse(comp_acmg$ClinVar_CLNREVSTAT == '_single_submitter',
                             '1',comp_acmg$stars)


comp_acmg %>% 
  separate(Uploaded_variation, into=c('x','y','REF','ALT'),sep = '_') %>%
  select(-x,-y) %>%
  relocate(Existing_variation, .before = REF) %>%
  write.table('comp_acmgFINAL.tsv',quote = F,col.names = T,sep='\t',row.names = F)

comp_acmg %>% 
  group_by(SYMBOL) %>%
  summarise(PL_AC = sum(PL_AC)) %>%
  ggplot(aes(y=reorder(SYMBOL,-PL_AC),x=PL_AC)) +
  geom_bar(fill="#48C095", stat = 'identity') + 
  theme_classic() +
  ylab('Gene') +
  xlab('Allele count') 
  
```


### 4. ClinVar variants

```{r clinvar, echo=FALSE, warning=FALSE}
clinsig <- read.table('../input/diseases/clin_sig_ready.tsv',sep='\t',header=T)
comp_clinsig <- clinsig %>% 
  filter(PL_AC > 0) %>%
   select(Uploaded_variation,Existing_variation,Location,starts_with('ClinVar'), 
          Gene, PL_AC, all_of(af_list)) %>%
  mutate(across(PL_AF:gnomADg_AF,as.character)) %>%
   mutate(across(PL_AF:gnomADg_AF,as.numeric)) 

comp_clinsig$stars <- NA
comp_clinsig$stars <- ifelse(comp_clinsig$ClinVar_CLNREVSTAT == 'practice_guideline',
                             '4',comp_clinsig$stars)
comp_clinsig$stars <- ifelse(comp_clinsig$ClinVar_CLNREVSTAT == 'reviewed_by_expert_panel',
                             '3',comp_clinsig$stars)
comp_clinsig$stars <- ifelse(
  comp_clinsig$ClinVar_CLNREVSTAT == 'criteria_provided,_multiple_submitters,_no_conflicts',
                             '2',comp_clinsig$stars)
comp_clinsig$stars <- ifelse(comp_clinsig$ClinVar_CLNREVSTAT == '_conflicting_interpretations',
                             '1',comp_clinsig$stars)
comp_clinsig$stars <- ifelse(comp_clinsig$ClinVar_CLNREVSTAT == '_single_submitter',
                             '1',comp_clinsig$stars)

comp_clinsig %>% filter(is.na(stars) ==F) %>%
  separate(Uploaded_variation, into=c('x','y','REF','ALT'),sep = '_') %>%
  select(-x,-y) %>%
  relocate(Existing_variation, .before = REF) %>%
  write.table('comp_clinsig.tsv',quote = F,col.names = T,sep='\t',row.names = F)

comp_clinsig %>% filter(is.na(stars) ==F) %>%
  select(Existing_variation,stars) %>% 
  distinct() %>%
  ggplot(aes(y=stars)) +
  geom_bar(fill="#48C095",width=0.5) + 
  theme_classic() +
  ylab('Number of gold stars') +
  xlab('Variants count') 

kable((comp_clinsig %>% 
        select(Existing_variation,stars) %>% 
        distinct() %>%
        group_by(stars) %>%
        count()))

```

### 5. Putative variants

```{r putative, echo=FALSE, warning=FALSE}
putative <- fread('../input/diseases/putative_ready.tsv',header=T)
comp_putative <- putative %>% 
   select(Uploaded_variation,Location,CLIN_SIG,ClinVar_CLNSIG,ClinVar_CLNREVSTAT,
          ClinVar_CLNDN, SYMBOL, Gene, PL_AC, all_of(af_list)) %>%
  mutate(across(PL_AF:gnomADg_AF,as.character)) %>%
   mutate(across(PL_AF:gnomADg_AF,as.numeric)) %>%
  filter(ClinVar_CLNSIG != '-') 

comp_putative %>% select(all_of(af_list)) %>%
  pivot_longer(everything(), names_to = 'group',values_to = 'AF') %>%
  filter(AF != 0) %>%
  ggplot(aes(x=group,y=log10(AF))) +
  geom_boxplot(fill="#48C095") +
  theme_classic() +
  xlab('') +
  ylab('log10 AF') +
  ggtitle('Allele frequency of putative variants') +
  theme(plot.title = element_text(hjust = 0.5))
  
glimpse(comp_putative)
```

### 6. % IMPACT variants

```{r echo=FALSE}
group.colors <- c(HIGH = "#27384A", MODERATE ="#48C095", LOW = "#B6B6B6")
stacked <- read.table('../input/diseases/impact_stacked_ready.tsv',sep='\t',header=T)
stacked$group <- factor(stacked$group)
stacked$group <- ordered(stacked$group, levels = c("<0.1%", "0.1-0.5%", ">0.5%"))

stacked %>%
ggplot(aes(fill=IMPACT,y=n,x=group)) +
geom_bar(position="fill", stat="identity") +
theme_classic() +
scale_fill_manual(values = group.colors) +
xlab('Allele frequencies') +
ylab('% of variants') 


```

### 7. Variants per functional category

```{r, echo=FALSE}
type.colors <- c(Exonic = "#27384A", Intronic ="#48C095", Noncoding = "#B6B6B6")
consequence <- read.table('../input/diseases/consequence_ready.tsv',header=T)

consequence$group <- factor(consequence$group)
consequence$group <- ordered(consequence$group, levels = c("<0.1%", "0.1-0.5%", ">0.5%"))

write.table(consequence,'consequence_summary.tsv',quote = F,col.names = T,sep='\t',row.names = F)

cons_list <- consequence %>% group_by(Consequence) %>% 
  summarise(n = sum(n))
  write.table(cons_list,'consequence_list.tsv', quote = F,col.names = T,sep='\t',row.names = F)

 consequence %>% 
  ggplot(aes(fill=type,y=n,x=group)) +
  geom_bar(position="fill", stat="identity") +
  theme_classic() +
  scale_fill_manual(values = type.colors) +
  xlab('Allele frequencies') +
  ylab('% of variants')

 
glimpse(consequence)
```

### 9. Variants per consequence

```{r, echo=FALSE}
consequence <- read.table('consequence_final_summary.tsv',header=T)
consequence$category <- gsub('_',' ',consequence$category)
consequence$Consequence <- gsub('_',' ',consequence$Consequence)

consequence %>% 
  ggplot(aes(x=n/1e+06,y=reorder(category,-n))) +
  geom_bar(stat="identity",fill="#48C095") +
  theme_classic() +
#  scale_fill_manual(values = type.colors) +
  xlab('Number of variants (millions)') +
  ylab('Consequence')

kable(
  consequence %>% group_by(category) %>% summarise(millions = sum(n/1e+06))
)

kable(consequence)


```
<!-- ## 9. Number of variants per impact -->

<!-- ```{r AF and IMPACT, echo=FALSE} -->
<!-- common <- read.table('../output/common_summary.tsv',header = T) -->
<!-- mediumrare <- read.table('../output/mediumrare_summary.tsv',header = T) -->
<!-- rare <- read.table('../output/rare_summary.tsv',header=T) -->
<!-- missing <- read.table('../output/missing_summary.tsv',header=T) -->

<!-- variants <- rbind(common,mediumrare,rare,missing) -->
<!-- kable(variants %>% pivot_wider(names_from = IMPACT, values_from = n, values_fill = 0) %>% -->
<!--         relocate(MODERATE, .before = MODIFIER) %>% filter(AF != '0%') -->
<!--         ) -->

<!-- ``` -->

## 10. NBS 

```{r NBS, echo=FALSE}
nbs <- read.table('nbs.tsv',header = T,sep='\t') 
rsid <- nbs$Existing_variation

nbs %>% 
  select(PL_AF, starts_with('gnomAD_')) %>%
  pivot_longer(everything(),names_to = 'pop',values_to = 'AF') %>%
  ggplot(aes(x=AF*100,y=reorder(pop,-AF))) +
  geom_bar(stat = 'identity',fill="#48C095") +
  theme_classic() +
  ylab('Population') +
  xlab('AF [%]') +
  geom_text(aes(label=paste(round(AF*100,2),'%'),
                hjust=-0.3)) +
  xlim(c(0,0.4))

```


## 11. Cystic fybrosis

```{r Mucoviscidosis, echo=FALSE}
muko <- read.table('muko_vep.tsv',header = T,sep='\t') 
rsid <- muko$Existing_variation

muko[1,] %>%
  select(PL_AF, starts_with('gnomAD_')) %>%
  pivot_longer(everything(),names_to = 'pop',values_to = 'AF') %>%
  ggplot(aes(x=AF*100,y=reorder(pop,-AF))) +
  geom_bar(stat = 'identity',fill="#48C095") +
  theme_classic() +
  ylab('Population') +
  xlab('AF [%]') +
    geom_text(aes(label=paste(round(AF*100,2),'%'),
                hjust=-0.3)) +
  xlim(c(0,2))



```





