---
title: "Variants in disease causing genes (1076 unrelated individuals)"
output: 
  github_document: default

---

```{r setup, echo=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo=FALSE, dev = 'png')
suppressMessages(library(knitr))
suppressMessages(library(tidyverse))
suppressMessages(library(data.table))
suppressMessages(library(viridis))
suppressMessages(library(gridExtra))
```

### Samples count

```{r echo=FALSE,warning=FALSE}
psc <- read.table('../input/psc.txt',sep='\t',header=F)
colnames(psc) <- c(
  'PSC','id','sample','nRefHom','nNonRefHom','nHets','nTransitions',
  'nTransversions','nIndels','average depth','nSingletons','nHapRef','nHapAlt','nMissing'
)

psc_summary <- psc %>% mutate(SNV = nNonRefHom + nHets) %>%
  select(sample,SNV,nIndels,nSingletons) %>%
  rename(Indels = nIndels,Singletons = nSingletons) %>%
  pivot_longer(-sample,names_to = 'Variant',values_to = 'count') %>%
  group_by(Variant) %>%
  summarise(min_count = min(count),mean_count = mean(count),max_count = max(count),
            
            ) %>%
  mutate_at(
    vars(ends_with('count')), funs(round(.)))


total_psc <- psc %>% select(nRefHom:nHets)  %>%
  mutate(var_sum=nNonRefHom+nHets)

  
kable(psc_summary)
```


## Cummulative allele frequency


```{r af_hist_pct,echo=FALSE,warning=FALSE}
af <- read.table('../input/af_stats.tsv')
colnames(af) <- c('AF','id', 'allele_frequency', 'SNV', 'number_of_transitions', 'number_of transversions', 'indel', 'repeat-consistent','repeat-inconsistent', 'not_applicable')

af <- af %>% mutate(allele_frequency = allele_frequency+0.000464684)
af_plot <- af %>% select(allele_frequency,SNV,indel)
af_plot$SNV <- 100*cumsum(af_plot$SNV)/sum(af_plot$SNV)
af_plot$indel <- 100*cumsum(af_plot$indel)/sum(af_plot$indel)


type.colors <- c(SNV = "#27384A", indel ="gray")

af_plot %>%
  pivot_longer(-allele_frequency,names_to = 'Variant class',
               values_to = 'Cummulative number of variants (millions)') %>%
  ggplot(aes(allele_frequency,`Cummulative number of variants (millions)`,
             linetype=`Variant class`)) +
  geom_line() +
  xlab('log10 Allele frequency') +
  #scale_color_manual(values = type.colors) +
  scale_y_continuous(breaks = seq(0,100,5)) +
  scale_x_log10() +
  theme_minimal()


```


```{r sv.af.hist, echo=F,warning=FALSE}
sv.af <- read.table('../../../genom_polaka/sv/sv_multisample_210716.smoove.square.unrelated.af_stats_PASS.tsv',header=T)
#colnames(sv.af) <- c('chrom','pos', 'svtype', 'len', 'ac', 'an')
#sv.af$af <- sv.af$ac/sv.af$an
sv.af <- sv.af[order(sv.af$af), ]

kable(sv.af %>% group_by(svtype) %>% summarise(n=n()))

colnames(af_plot)[1] <- 'af'

af_perc_plot <- af_plot %>%
  pivot_longer(-af,names_to = 'svtype',values_to = 'cspct') %>%
  relocate(svtype, .before = af) %>%
  mutate(svtype = as.factor(svtype))


sv.af.cs  <-  sv.af %>%
             group_by(svtype, af) %>%
             summarise(cnt=n()) %>%
             group_by(svtype) %>%
             summarise(af, cs=cumsum(cnt), cspct=100*cumsum(cnt)/sum(cnt)) %>%
  select(-cs) %>% bind_rows(af_perc_plot)

sv.af.cs$svtype[sv.af.cs$svtype=='SNP'] <- 'SNV'

# group.colors <- c(BND = "#BC0020", DEL = "#27384A", DUP ="gray", INV = "blue",
#                   SNP = 'darkgoldenrod4', indel='#5F5F5F')

sv.af.cs %>% filter(svtype != 'BND') %>%
ggplot() +
  geom_line(aes(x = af, y=cspct, linetype=svtype)) +
  scale_x_log10()+
  xlab('log10 Allele frequency') + ylab('% variants') +
  #scale_color_viridis(discrete = T, 'Coding consequence') +
  theme_minimal() +
  theme(legend.position = 'top') +
  guides(linetype = guide_legend(override.aes = list(size = 0.4, width=50)))
  

```

```{r sv.af.hist.col, echo=F,warning=FALSE}
sv.af.cs %>% filter(svtype != 'BND') %>%
ggplot() +
  geom_line(aes(x = af, y=cspct, col=svtype),size=1) +
  scale_x_log10()+
  xlab('log10 Allele frequency') + ylab('% variants') +
  theme_minimal() +
  theme(legend.position = 'top') +
  guides(linetype = guide_legend(override.aes = list(size = 0.4, width=50)))
```


```{r, echo=FALSE, warning=FALSE}
af_list <- c('PL_AF',  'gnomADg_AF_NFE','gnomADg_AF')
group.colors <- c(PL_AF = "#BC0020", AF = "#27384A", gnomAD_FIN_AF ="gray", gnomAD_AF_NFE = "#B6B6B6",EUR_AF='#5F5F5F')

```

## 3. ACMG

```{r ACMG, echo=FALSE, warning=FALSE}
acmg <- read.table('../input/diseases/acmg_ready.tsv',sep='\t',header=T)
comp_acmg <- acmg %>% 
   select(Uploaded_variation, Existing_variation,Location,SYMBOL,Gene,PUBMED,Codons,
          ClinVar_CLNSIG,ClinVar_CLNREVSTAT,ClinVar,IMPACT,
          stars,PL_AC, gnomAD3g_AF,PL_AF) %>%
  mutate(across(gnomAD3g_AF,as.character)) %>%
   mutate(across(gnomAD3g_AF,as.numeric)) 
comp_acmg %>% 
  separate(Uploaded_variation, into=c('x','y','REF','ALT'),sep = '_') %>%
  select(-x,-y) %>%
  relocate(Existing_variation, .before = REF) %>%
  write.table('comp_acmgFINAL.tsv',quote = F,col.names = T,sep='\t',row.names = F)

comp_acmg %>% 
  ggplot(aes(y=reorder(SYMBOL,-PL_AC),x=PL_AC)) +
  geom_bar(fill="gray", stat = 'identity') + 
  theme_classic() +
  ylab('Gene') +
  xlab('Allele count') 
  
```


### ClinVar variants pathogenic & likely pathogenic \ filtered with gnomad3g AF < 0.001

```{r clinvar_001, echo=FALSE, warning=FALSE}
clinsig <- fread('../input/diseases/clin_sig_ready.tsv',header=T) 
clinsig$gnomAD3g_AF <- ifelse(clinsig$gnomAD3g_AF == '-',0,clinsig$gnomAD3g_AF)
clinsig$gnomAD3g_AF <-as.numeric(clinsig$gnomAD3g_AF)

clinsig <- clinsig %>%
  filter(gnomAD3g_AF < 0.001 & is.na(stars) == F) 

clinsig %>% na.omit() %>%
  separate(Uploaded_variation, into=c('x','y','REF','ALT'),sep = '_') %>%
  select(-x,-y,-Allele) %>%
  relocate(Existing_variation, .before = REF) %>%
  arrange(desc(gnomAD3g_AF)) %>%
  write.table('comp_clinsig001.tsv',quote = F,col.names = T,sep='\t',row.names = F)

clinsig %>% na.omit() %>%
  select(Existing_variation,stars) %>% 
  distinct() %>%
  ggplot(aes(y=factor(stars))) +
  geom_bar(fill="gray",width=0.5) + 
  theme_classic() +
  ylab('Number of gold stars') +
  xlab('Variants count') 

kable((clinsig %>% na.omit() %>%
        select(Existing_variation,stars) %>% 
        group_by(stars) %>%
        count()),caption = 'variants with Clinvar stars')

```

<!-- ### ClinVar variants pathogenic & likely pathogenic \ filtered with gnomad3g AF < 0.01 -->

<!-- ```{r clinvar_01, echo=FALSE, warning=FALSE} -->
<!-- clinsig <- fread('../input/diseases/clin_sig_ready.tsv',header=T)  -->
<!-- clinsig$gnomAD3g_AF <- ifelse(clinsig$gnomAD3g_AF == '-',0,clinsig$gnomAD3g_AF) -->
<!-- clinsig$gnomAD3g_AF <-as.numeric(clinsig$gnomAD3g_AF) -->

<!-- clinsig <- clinsig %>% -->
<!--   filter(gnomAD3g_AF < 0.01) %>% na.omit() -->

<!-- clinsig %>% na.omit() %>% -->
<!--   separate(Uploaded_variation, into=c('x','y','REF','ALT'),sep = '_') %>% -->
<!--   select(-x,-y,-Allele) %>% -->
<!--   relocate(Existing_variation, .before = REF) %>% -->
<!--   arrange(desc(gnomAD3g_AF)) %>% -->
<!--   write.table('comp_clinsig01.tsv',quote = F,col.names = T,sep='\t',row.names = F) -->

<!-- clinsig %>% na.omit() %>% -->
<!--   select(Existing_variation,stars) %>%  -->
<!--   distinct() %>% -->
<!--   ggplot(aes(y=factor(stars))) + -->
<!--   geom_bar(fill="gray",width=0.5) +  -->
<!--   theme_classic() + -->
<!--   ylab('Number of gold stars') + -->
<!--   xlab('Variants count')  -->

<!-- kable((clinsig %>% na.omit() %>% -->
<!--         select(Existing_variation,stars) %>%  -->
<!--         group_by(stars) %>% -->
<!--         count()),caption = 'variants with Clinvar stars') -->


<!-- ``` -->



<!-- ### Putative variants -->

<!-- ```{r putative, echo=FALSE, warning=FALSE} -->
<!-- putative <- fread('../input/diseases/putative_ready.tsv',header=T) -->
<!-- comp_putative <- putative %>%  -->
<!--    select(Uploaded_variation,Location,CLIN_SIG,ClinVar_CLNSIG,ClinVar_CLNREVSTAT, -->
<!--           ClinVar_CLNDN, SYMBOL, Gene, PL_AC, all_of(af_list)) %>% -->
<!--   mutate(across(PL_AF:gnomADg_AF,as.character)) %>% -->
<!--    mutate(across(PL_AF:gnomADg_AF,as.numeric)) %>% -->
<!--   filter(ClinVar_CLNSIG != '-')  -->

<!-- ``` -->

<!-- ### % IMPACT variants -->

<!-- ```{r echo=FALSE,warning=FALSE} -->
<!-- group.colors <- c(HIGH = "#27384A", MODERATE ="gray", LOW = "#B6B6B6") -->
<!-- stacked <- read.table('../input/diseases/impact_stacked_ready.tsv',sep='\t',header=T) -->
<!-- stacked$group <- factor(stacked$group) -->
<!-- stacked$group <- ordered(stacked$group, levels = c("<0.1%", "0.1-0.5%", ">0.5%")) -->

<!-- stacked %>% -->
<!-- ggplot(aes(fill=IMPACT,y=n,x=group)) + -->
<!-- geom_bar(position="fill", stat="identity") + -->
<!-- theme_classic() + -->
<!-- scale_fill_manual(values = group.colors) + -->
<!-- xlab('Allele frequencies') + -->
<!-- ylab('% of variants')  -->


<!-- ``` -->



## Number of variants per impact

```{r AF and IMPACT, echo=FALSE}
common <- read.table('../output/common_summary.tsv',header = T)
mediumrare <- read.table('../output/mediumrare_summary.tsv',header = T)
rare <- read.table('../output/rare_summary.tsv',header=T)
#missing <- read.table('../output/missing_summary.tsv',header=T)
variants <- rbind(common,mediumrare,rare) %>% filter(VARIANT_CLASS != 'indel')
kable(variants %>% pivot_wider(names_from = IMPACT, values_from = n, values_fill = 0) %>%
        relocate(MODERATE, .before = MODIFIER) %>% filter(AF != '0%')
        )

variants %>% pivot_wider(names_from = IMPACT, values_from = n, values_fill = 0) %>%
        relocate(MODERATE, .before = MODIFIER) %>% filter(AF != '0%') %>%
  write.table('variants_class.tsv',quote=F,col.names = T,row.names = F,sep='\t')

kable(variants %>% group_by(IMPACT) %>% summarise(n=sum(n)))
kable(variants %>% group_by(IMPACT,AF) %>% summarise(n=sum(n)))
```

### Variants per coding consequence

```{r, echo=FALSE,warning=F}
consequence_list <- read.table('../input/diseases/consequence_list.csv',header=T,
                          sep=',') %>% select(-n)

consequence <- read.table('../input/diseases/consequence_ready.tsv',header=T,
                          sep = '\t') %>% 
  left_join(consequence_list,by='Consequence') %>%
  mutate(Coding_var_category = ifelse(is.na(Coding_var_category)==T,'nonexonic',Coding_var_category)) %>%
  mutate(Coding_var_category = ifelse(Coding_var_category == 'splice_region','splicing_variant',
                                      Coding_var_category))


consequence$group <- factor(
  consequence$group, levels = c('0 - 0.1%','0.1 - 0.2%', '0.2 - 0.5%',
                                '0.5 - 1%','1 - 2%','2 - 5%','5 - 10%',
                                '10 - 50%','50 - 100%'))

consequence$Konsekwencje <- consequence$Coding_var_category
cons_level <-list('3utr_variant',
                                        '5utr_variant',
                                        'frameshift_variant',
                                        'inframe_indel',
                                        'missense_variant',
                                        'nonexonic',
                                        'other',
                                        'splicing_variant',
                                        'start_stop_variant',
                                        'synonymous_variant')

consequence$Konsekwencje[consequence$Konsekwencje == '3utr_variant'] <- "wariant w regionie 3'UTR"
consequence$Konsekwencje[consequence$Konsekwencje == '5utr_variant'] <- "wariant w regionie 5'UTR"
consequence$Konsekwencje[consequence$Konsekwencje == 'frameshift_variant'] <- "przesunięcie ramki odczytu"
consequence$Konsekwencje[consequence$Konsekwencje == 'inframe_indel'] <- "insercja/delecja"
consequence$Konsekwencje[consequence$Konsekwencje == 'missense_variant'] <- "mutacja missensowna"
consequence$Konsekwencje[consequence$Konsekwencje == 'nonexonic'] <- "wariant niekodujący"
consequence$Konsekwencje[consequence$Konsekwencje == 'other'] <- "inne"
consequence$Konsekwencje[consequence$Konsekwencje == 'splicing_variant'] <- "wariant splicingowy"
consequence$Konsekwencje[consequence$Konsekwencje == 'start_stop_variant'] <- "wariant w kodonie start/stop"
consequence$Konsekwencje[consequence$Konsekwencje == 'synonymous_variant'] <- "wariant synonimiczny"

consequence$Coding_var_category <- gsub('_',' ',consequence$Coding_var_category)

cons_plot <- consequence %>% group_by(Coding_var_category, group) %>%
  summarise(n = sum(n)) %>%
  mutate(percentage = n/(sum(n)),
         minmax = (n - min(n))/(max(n)-min(n))
           )

cons_plot_pl <- consequence %>%
  group_by(Konsekwencje, group) %>%
  summarise(n = sum(n)) %>%
  mutate(percentage = n/(sum(n)),
         minmax = (n - min(n))/(max(n)-min(n)),
         Konsewkencje = factor(Konsekwencje, levels = cons_level)
           ) 

# cons_plot_pl %>% filter(Konsekwencje != 'inne') %>% 
#   ggplot(aes(x=as.numeric(group),y=percentage,fill=Konsekwencje)) +
#   geom_area(alpha=0.6 , size=.5, colour="white",position = 'fill') +
#     scale_fill_viridis(discrete = T,'Typ wariantu', ) +
#   theme_minimal() +
#   scale_x_continuous(breaks = seq(1,9,2), labels = unique(cons_plot$group)[seq(1,9,2)]) +
#   scale_y_continuous(breaks = seq(0,1,0.25),labels = c('0%','25%','50%','75%','100%')) +
# xlab('Przedziały częstości alleli') +
#   ylab('Procentowy udział wariantów w danym przedziale')

cons_plot %>% filter(Coding_var_category != 'other') %>% 
  ggplot(aes(x=as.numeric(group),y=percentage,fill=Coding_var_category)) +
  geom_area(alpha=0.6 , size=.5, colour="white",position = 'fill') +
    scale_fill_viridis(discrete = T,'Coding consequence') +
  theme_minimal() +
  scale_x_continuous(breaks = seq(1,9,2), labels = unique(cons_plot$group)[seq(1,9,2)]) +
  scale_y_continuous(breaks = seq(0,1,0.25),labels = c('0%','25%','50%','75%','100%')) +
xlab('Allele frequency') +
  ylab('Normalized percentage')

```

### Variants per non-coding consequence

```{r non-coding consequence, echo=FALSE}
consequence$noncoding.var_category <- gsub('_',' ',consequence$noncoding.var_category)

cons_plot <- consequence %>% group_by(noncoding.var_category, group) %>%
  summarise(n = sum(n)) %>%
  mutate(percentage = n/(sum(n)),
         minmax = (n - min(n))/(max(n)-min(n))
           )

cons_plot %>%
  ggplot(aes(x=as.numeric(group),y=percentage,fill=noncoding.var_category)) +
  geom_area(alpha=0.6 , size=.5, colour="white",position = 'fill') +
    scale_fill_viridis(discrete = T, 'Coding consequence') +
  theme_minimal() +
  scale_x_continuous(breaks = seq(1,9,2), labels = unique(cons_plot$group)[seq(1,9,2)]) +
xlab('Allele frequency') +
  ylab('Normalized percentage')

```

```{r echo=F}
variants_df <- read.csv('variants_gnomad31.csv',header=T)
```

## NBS: chr8_89971213_ATTTGT_A 

```{r NBS, echo=FALSE}
nbs <- read.table('nbs.tsv',header = T,sep='\t') 
nbs <- nbs %>% 
  select(PL_AF, starts_with('gnomAD_')) %>%
  pivot_longer(everything(),names_to = 'pop',values_to = 'AF') %>% 
  filter(pop == 'PL_AF') 
nbs <- variants_df %>% filter(variant=='NBS') %>% 
  bind_rows(nbs) %>%
  select(-variant)
nbs$pop <- ifelse(nbs$pop == 'PL_AF','POL',nbs$pop)
nbs$pop <- gsub('_',' ',nbs$pop)
nbs_plot <- nbs %>% 
  ggplot(aes(x=AF*100,y=reorder(pop,-AF))) +
  geom_bar(stat = 'identity',fill="gray") +
  theme_classic() +
  ylab('Population') +
  xlab('AF [%]') +
  geom_text(aes(label=paste(round(AF*100,2),'%'),
                hjust=-0.3)) +
  xlim(c(0,0.4)) +
  theme(axis.text = element_text(size=15))
nbs_plot
```


## Cystic fybrosis: chr7_117559590_ATCT_A

```{r Mucoviscidosis, echo=FALSE}
muko <- read.table('muko_vep.tsv',header = T,sep='\t') 
muko <- muko %>% 
  select(PL_AF, starts_with('gnomAD_')) %>%
  pivot_longer(everything(),names_to = 'pop',values_to = 'AF') %>% 
  filter(pop == 'PL_AF')  %>% distinct()
muko <- variants_df %>% filter(variant=='CFTR') %>% 
  bind_rows(muko) %>%
  select(-variant)
muko$pop <- ifelse(muko$pop == 'PL_AF','POL',muko$pop)
muko$pop <- gsub('_',' ',muko$pop)
muko_plot <- muko %>% 
  ggplot(aes(x=AF*100,y=reorder(pop,-AF))) +
  geom_bar(stat = 'identity',fill="gray") +
  theme_classic() +
  ylab('Population') +
  xlab('AF [%]') +
  geom_text(aes(label=paste(round(AF*100,2),'%'),
                hjust=-0.3)) +
  xlim(c(0,2)) +
  theme(axis.text = element_text(size=15))
muko_plot
```



## DHCR7

```{r DHCR7, echo=FALSE}
dh <- clinsig %>% filter(grepl('rs11555217',Existing_variation))
dh <- dh %>% 
  select(PL_AF, starts_with('gnomAD_')) %>%
  mutate_if(is.character,as.numeric) %>%
  pivot_longer(everything(),names_to = 'pop',values_to = 'AF') %>% 
  filter(pop == 'PL_AF')  %>% distinct()
dh <- variants_df %>% filter(variant=='DHCR7') %>% 
  bind_rows(dh) %>%
  select(-variant)
dh$pop <- ifelse(dh$pop == 'PL_AF','POL',dh$pop)
dh$pop <- gsub('_',' ',dh$pop)
dh_plot <- dh %>% 
  ggplot(aes(x=AF*100,y=reorder(pop,-AF))) +
  geom_bar(stat = 'identity',fill="gray") +
  theme_classic() +
  ylab('Population') +
  xlab('AF [%]') +
  geom_text(aes(label=paste(round(AF*100,2),'%'),
                hjust=-0.3)) +
  xlim(c(0,2)) +
  theme(axis.text = element_text(size=15))
dh_plot
```