---
title: "Variants in disease causing genes \n Results for 943 unrelated individuals"
subtitle:
output: 
  github_document: default

---

```{r setup, echo=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo=FALSE, dev = 'jpeg')
suppressMessages(library(knitr))
suppressMessages(library(tidyverse))
suppressMessages(library(data.table))
suppressMessages(library(viridis))
suppressMessages(library(gridExtra))
```

### Samples count

```{r echo=FALSE,warning=FALSE}
psc <- read.table('../input/psc.txt',sep='\t',header=T)
colnames(psc) <- c(
  'PSC','id','sample','nRefHom','nNonRefHom','nHets','nTransitions',
  'nTransversions','nIndels','average depth','nSingletons','nHapRef','nHapAlt','nMissing'
)

psc_summary <- psc %>% mutate(SNP = nNonRefHom + nHets) %>%
  select(sample,SNP,nIndels,nSingletons) %>%
  rename(Indels = nIndels,Singletons = nSingletons) %>%
  pivot_longer(-sample,names_to = 'Variant',values_to = 'count') %>%
  group_by(Variant) %>%
  summarise(min_count = min(count),mean_count = mean(count),max_count = max(count),
            
            ) %>%
  mutate_at(
    vars(ends_with('count')), funs(round(.)))
  
kable(psc_summary)
```


## Cummulative allele frequency


```{r af_hist_pct,echo=FALSE,warning=FALSE}
af <- read.table('../input/af_stats.tsv')
colnames(af) <- c('AF','id', 'allele_frequency', 'SNP', 'number_of_transitions', 'number_of transversions', 'indel', 'repeat-consistent','repeat-inconsistent', 'not_applicable')

af$allele_frequency[1] <- 0.000530223
af_plot <- af %>% select(allele_frequency,SNP,indel)
af_plot$SNP <- 100*cumsum(af_plot$SNP)/sum(af_plot$SNP)
af_plot$indel <- 100*cumsum(af_plot$indel)/sum(af_plot$indel)


type.colors <- c(SNP = "#27384A", indel ="#48C095")

af_plot %>%
  pivot_longer(-allele_frequency,names_to = 'Variant class',
               values_to = 'Cummulative number of variants (millions)') %>%
  ggplot(aes(allele_frequency,`Cummulative number of variants (millions)`,
             col=`Variant class`)) +
  geom_line() +
  xlab('log10 Allele frequency') +
  scale_color_manual(values = type.colors) +
  scale_y_continuous(breaks = seq(0,100,5)) +
  scale_x_log10() +
  theme_minimal()


```


```{r sv.af.hist, echo=F,warning=FALSE}
sv.af <- read.table('../../../genom_polaka/sv/sv_multisample_210519.smoove.square.unrelated.nogt.stats.tsv')
colnames(sv.af) <- c('chrom','pos', 'svtype', 'len', 'ac', 'an')
sv.af$af <- sv.af$ac/sv.af$an
sv.af <- sv.af[order(sv.af$af), ]

colnames(af_plot)[1] <- 'af'

af_perc_plot <- af_plot %>% 
  pivot_longer(-af,names_to = 'svtype',values_to = 'cspct') %>%
  relocate(svtype, .before = af) %>% 
  mutate(svtype = as.factor(svtype))

sv.af.cs  <-  sv.af %>% 
             group_by(svtype, af) %>%
             summarise(cnt=n()) %>% 
             group_by(svtype) %>% 
             summarise(af, cs=cumsum(cnt), cspct=100*cumsum(cnt)/sum(cnt)) %>%
  select(-cs) %>% bind_rows(af_perc_plot)
  
group.colors <- c(BND = "#BC0020", DEL = "#27384A", DUP ="#48C095", INV = "blue",
                  SNP = 'darkgoldenrod4', indel='#5F5F5F')

sv.af.cs %>%
ggplot() + 
  geom_line(aes(x = af, y=cspct, color=svtype)) +
  scale_x_log10()+
  xlab('log10 Allele frequency') + ylab('% variants') +
  scale_color_viridis(discrete = T, 'Coding consequence') +
  theme_minimal() +
  theme(legend.position = 'top')


```

```{r, echo=FALSE, warning=FALSE}
af_list <- c('PL_AF',  'gnomADg_AF_NFE','gnomADg_AF')
group.colors <- c(PL_AF = "#BC0020", AF = "#27384A", gnomAD_FIN_AF ="#48C095", gnomAD_AF_NFE = "#B6B6B6",EUR_AF='#5F5F5F')

```

## ACMG

```{r ACMG, echo=FALSE, warning=FALSE}
acmg <- read.table('../input/diseases/acmg_ready.tsv',sep='\t',header=T)

comp_acmg <- acmg %>% 
    filter(PL_AC > 0) %>%
   select(Uploaded_variation,Existing_variation,Location,starts_with('ClinVar'), 
          SYMBOL,Gene, PL_AC, all_of(af_list)) %>%
  mutate(across(PL_AF:gnomADg_AF,as.character)) %>%
   mutate(across(PL_AF:gnomADg_AF,as.numeric))

comp_acmg$stars <- NA
comp_acmg$stars <- ifelse(comp_acmg$ClinVar_CLNREVSTAT == 'practice_guideline',
                             '4',comp_acmg$stars)
comp_acmg$stars <- ifelse(comp_acmg$ClinVar_CLNREVSTAT == 'reviewed_by_expert_panel',
                             '3',comp_acmg$stars)
comp_acmg$stars <- ifelse(
  comp_acmg$ClinVar_CLNREVSTAT == 'criteria_provided,_multiple_submitters,_no_conflicts',
                             '2',comp_acmg$stars)
comp_acmg$stars <- ifelse(comp_acmg$ClinVar_CLNREVSTAT == '_conflicting_interpretations',
                             '1',comp_acmg$stars)
comp_acmg$stars <- ifelse(comp_acmg$ClinVar_CLNREVSTAT == '_single_submitter',
                             '1',comp_acmg$stars)


comp_acmg %>% 
  separate(Uploaded_variation, into=c('x','y','REF','ALT'),sep = '_') %>%
  select(-x,-y) %>%
  relocate(Existing_variation, .before = REF) %>%
  write.table('comp_acmgFINAL.tsv',quote = F,col.names = T,sep='\t',row.names = F)

comp_acmg %>% 
  group_by(SYMBOL) %>%
  summarise(PL_AC = sum(PL_AC)) %>%
  ggplot(aes(y=reorder(SYMBOL,-PL_AC),x=PL_AC)) +
  geom_bar(fill="#48C095", stat = 'identity') + 
  theme_classic() +
  ylab('Gene') +
  xlab('Allele count') 
  
```


### ClinVar variants pathogenic & likely pathogenic \ filtered with gnomad3g AF < 0.001

```{r clinvar_001, echo=FALSE, warning=FALSE}
clinsig <- fread('../input/diseases/clin_sig_ready.tsv',header=T) 
clinsig$gnomAD3g_AF <- ifelse(clinsig$gnomAD3g_AF == '-',0,clinsig$gnomAD3g_AF)
clinsig$gnomAD3g_AF <-as.numeric(clinsig$gnomAD3g_AF)

clinsig <- clinsig %>%
  filter(gnomAD3g_AF < 0.001) %>% na.omit()

clinsig %>% na.omit() %>%
  separate(Uploaded_variation, into=c('x','y','REF','ALT'),sep = '_') %>%
  select(-x,-y,-Allele) %>%
  relocate(Existing_variation, .before = REF) %>%
  arrange(desc(gnomAD3g_AF)) %>%
  write.table('comp_clinsig001.tsv',quote = F,col.names = T,sep='\t',row.names = F)

clinsig %>% na.omit() %>%
  select(Existing_variation,stars) %>% 
  distinct() %>%
  ggplot(aes(y=factor(stars))) +
  geom_bar(fill="#48C095",width=0.5) + 
  theme_classic() +
  ylab('Number of gold stars') +
  xlab('Variants count') 

kable((clinsig %>% na.omit() %>%
        select(Existing_variation,stars) %>% 
        group_by(stars) %>%
        count()),caption = 'variants with Clinvar stars')

```

### ClinVar variants pathogenic & likely pathogenic \ filtered with gnomad3g AF < 0.01

```{r clinvar_01, echo=FALSE, warning=FALSE}
clinsig <- fread('../input/diseases/clin_sig_ready.tsv',header=T) 
clinsig$gnomAD3g_AF <- ifelse(clinsig$gnomAD3g_AF == '-',0,clinsig$gnomAD3g_AF)
clinsig$gnomAD3g_AF <-as.numeric(clinsig$gnomAD3g_AF)

clinsig <- clinsig %>%
  filter(gnomAD3g_AF < 0.01) %>% na.omit()

clinsig %>% na.omit() %>%
  separate(Uploaded_variation, into=c('x','y','REF','ALT'),sep = '_') %>%
  select(-x,-y,-Allele) %>%
  relocate(Existing_variation, .before = REF) %>%
  arrange(desc(gnomAD3g_AF)) %>%
  write.table('comp_clinsig01.tsv',quote = F,col.names = T,sep='\t',row.names = F)

clinsig %>% na.omit() %>%
  select(Existing_variation,stars) %>% 
  distinct() %>%
  ggplot(aes(y=factor(stars))) +
  geom_bar(fill="#48C095",width=0.5) + 
  theme_classic() +
  ylab('Number of gold stars') +
  xlab('Variants count') 

kable((clinsig %>% na.omit() %>%
        select(Existing_variation,stars) %>% 
        group_by(stars) %>%
        count()),caption = 'variants with Clinvar stars')


```



### Putative variants

```{r putative, echo=FALSE, warning=FALSE}
putative <- fread('../input/diseases/putative_ready.tsv',header=T)
comp_putative <- putative %>% 
   select(Uploaded_variation,Location,CLIN_SIG,ClinVar_CLNSIG,ClinVar_CLNREVSTAT,
          ClinVar_CLNDN, SYMBOL, Gene, PL_AC, all_of(af_list)) %>%
  mutate(across(PL_AF:gnomADg_AF,as.character)) %>%
   mutate(across(PL_AF:gnomADg_AF,as.numeric)) %>%
  filter(ClinVar_CLNSIG != '-') 

```

### % IMPACT variants

```{r echo=FALSE,warning=FALSE}
group.colors <- c(HIGH = "#27384A", MODERATE ="#48C095", LOW = "#B6B6B6")
stacked <- read.table('../input/diseases/impact_stacked_ready.tsv',sep='\t',header=T)
stacked$group <- factor(stacked$group)
stacked$group <- ordered(stacked$group, levels = c("<0.1%", "0.1-0.5%", ">0.5%"))

stacked %>%
ggplot(aes(fill=IMPACT,y=n,x=group)) +
geom_bar(position="fill", stat="identity") +
theme_classic() +
scale_fill_manual(values = group.colors) +
xlab('Allele frequencies') +
ylab('% of variants') 


```



## Number of variants per impact

```{r AF and IMPACT, echo=FALSE}
common <- read.table('../output/common_summary.tsv',header = T)
mediumrare <- read.table('../output/mediumrare_summary.tsv',header = T)
rare <- read.table('../output/rare_summary.tsv',header=T)
#missing <- read.table('../output/missing_summary.tsv',header=T)
variants <- rbind(common,mediumrare,rare)
kable(variants %>% pivot_wider(names_from = IMPACT, values_from = n, values_fill = 0) %>%
        relocate(MODERATE, .before = MODIFIER) %>% filter(AF != '0%')
        )

variants %>% pivot_wider(names_from = IMPACT, values_from = n, values_fill = 0) %>%
        relocate(MODERATE, .before = MODIFIER) %>% filter(AF != '0%') %>%
  write.table('variants_class.tsv',quote=F,col.names = T,row.names = F,sep='\t')
```

### Variants per coding consequence

```{r, echo=FALSE,warning=F}
consequence <- read.table('plik_do_wykresu_consequence_data.tsv',header=T,
                          sep='\t')

consequence$group <- factor(
  consequence$group, levels = c('0 - 0.1%','0.1 - 0.2%', '0.2 - 0.5%',
                                '0.5 - 1%','1 - 2%','2 - 5%','5 - 10%',
                                '10 - 50%','50 - 100%'))

consequence$Konsekwencje <- consequence$Coding_var_category
levels(consequence$Konsekwencje) <-list(`wariant w regionie 3'UTR` = '3utr_variant', 
                                        `wariant w regionie 5'UTR` = '5utr_variant',
                                        `przesunięcie ramki odczytu` = 'frameshift_variant',
                                        `insercja/delecja` = 'inframe_indel',
                                        `wariant synonimiczny` = 'missense_variant',
                                        `wariant niekodujący` = 'nonexonic',
                                        inne = 'other',
                                        `wariant splicingowy`='splicing_variant',
                                        `wariant w kodonie start/stop` = 'start_stop_variant',
                                        `wariant synonimiczny` = 'synonymous_variant')

consequence$Coding_var_category <- gsub('_',' ',consequence$Coding_var_category)

cons_plot <- consequence %>% group_by(Coding_var_category, group) %>%
  summarise(n = sum(n)) %>% 
  mutate(percentage = n/(sum(n)),
         minmax = (n - min(n))/(max(n)-min(n))
           )

cons_plot_pl <- consequence %>% group_by(Konsekwencje, group) %>%
  summarise(n = sum(n)) %>% 
  mutate(percentage = n/(sum(n)),
         minmax = (n - min(n))/(max(n)-min(n))
           )

cons_plot_pl %>% filter(Konsekwencje != 'inne') %>%
  ggplot(aes(x=as.numeric(group),y=percentage,fill=Konsekwencje)) +
  geom_area(alpha=0.6 , size=.5, colour="white",position = 'fill') +
    scale_fill_viridis(discrete = T,'Typ wariantu') +
  theme_minimal() +
  scale_x_continuous(breaks = seq(1,9,2), labels = unique(cons_plot$group)[seq(1,9,2)]) +
  scale_y_continuous(breaks = seq(0,1,0.25),labels = c('0%','25%','50%','75%','100%')) +
xlab('Przedziały częstości alleli') +
  ylab('Procentowy udział wariantów w danym przedziale') 

cons_plot %>% filter(Coding_var_category != 'other') %>%
  ggplot(aes(x=as.numeric(group),y=percentage,fill=Coding_var_category)) +
  geom_area(alpha=0.6 , size=.5, colour="white",position = 'fill') +
    scale_fill_viridis(discrete = T,'Coding consequence') +
  theme_minimal() +
  scale_x_continuous(breaks = seq(1,9,2), labels = unique(cons_plot$group)[seq(1,9,2)]) +
  scale_y_continuous(breaks = seq(0,1,0.25),labels = c('0%','25%','50%','75%','100%')) +
xlab('Allele frequency') +
  ylab('Normalized percentage') 

```

### Variants per non-coding consequence

```{r non-coding consequence, echo=FALSE}
consequence$noncoding.var_category <- gsub('_',' ',consequence$noncoding.var_category)

cons_plot <- consequence %>% group_by(noncoding.var_category, group) %>%
  summarise(n = sum(n)) %>% 
  mutate(percentage = n/(sum(n)),
         minmax = (n - min(n))/(max(n)-min(n))
           )

cons_plot %>% 
  ggplot(aes(x=as.numeric(group),y=percentage,fill=noncoding.var_category)) +
  geom_area(alpha=0.6 , size=.5, colour="white",position = 'fill') +
    scale_fill_viridis(discrete = T, 'Coding consequence') +
  theme_minimal() +
  scale_x_continuous(breaks = seq(1,9,2), labels = unique(cons_plot$group)[seq(1,9,2)]) +
xlab('Allele frequency') +
  ylab('Normalized percentage') 

```



## NBS: chr8_89971213_ATTTGT_A 

```{r NBS, echo=FALSE}
nbs <- read.table('nbs.tsv',header = T,sep='\t') 
rsid <- nbs$Existing_variation

nbs %>% 
  select(PL_AF, starts_with('gnomAD_')) %>%
  pivot_longer(everything(),names_to = 'pop',values_to = 'AF') %>%
  ggplot(aes(x=AF*100,y=reorder(pop,-AF))) +
  geom_bar(stat = 'identity',fill="#48C095") +
  theme_classic() +
  ylab('Population') +
  xlab('AF [%]') +
  geom_text(aes(label=paste(round(AF*100,2),'%'),
                hjust=-0.3)) +
  xlim(c(0,0.4))

```


## Cystic fybrosis: chr7_117559590_ATCT_A

```{r Mucoviscidosis, echo=FALSE}
muko <- read.table('muko_vep.tsv',header = T,sep='\t') 
rsid <- muko$Existing_variation

muko[1,] %>%
  select(PL_AF, starts_with('gnomAD_')) %>%
  pivot_longer(everything(),names_to = 'pop',values_to = 'AF') %>%
  ggplot(aes(x=AF*100,y=reorder(pop,-AF))) +
  geom_bar(stat = 'identity',fill="#48C095") +
  theme_classic() +
  ylab('Population') +
  xlab('AF [%]') +
    geom_text(aes(label=paste(round(AF*100,2),'%'),
                hjust=-0.3)) +
  xlim(c(0,0.5))
```

<!-- ## CFTR deletions -->

<!-- ```{r CFTR, echo=FALSE} -->
<!-- muko <- read.table('CFTR.tsv',header = T,sep='\t') %>% -->
<!--   select(Uploaded_variation,PL_AF, starts_with('gnomAD_'),VARIANT_CLASS) -->

<!-- dels <- muko %>% filter(VARIANT_CLASS == 'deletion', gnomAD_AF != '-')  -->
<!-- dels[,3:11] <- dels[,3:11] %>% mutate_if(is.factor,as.character)  -->
<!-- dels[,3:11] <- dels[,3:11] %>% mutate_if(is.character,as.numeric)  -->

<!-- dels %>% select(-VARIANT_CLASS)  %>% -->
<!--   pivot_longer(-Uploaded_variation,names_to = 'pop',values_to = 'AF') %>% -->
<!--   ggplot(aes(x=AF*100,y=reorder(pop,-AF),fill=Uploaded_variation)) + -->
<!--   geom_bar(stat = 'identity',position = 'fill') + -->
<!--   scale_fill_viridis(discrete = T, 'Variant') + -->
<!--   theme_classic() + -->
<!--   ylab('Population') + -->
<!--   xlab('AF [%]')  -->

<!-- kable(dels %>% select(-VARIANT_CLASS)) -->
<!-- ``` -->





