---
title: "Variants in disease causing genes"
output: 
  pdf_document: default
  github_document: default

---

```{r setup, echo=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo=FALSE)
suppressMessages(library(knitr))
suppressMessages(library(tidyverse))
suppressMessages(library(data.table))
library(grid)
suppressMessages(library(gridExtra))
```



## Variants distribution
```{r af_hist,echo=FALSE}
library(png)
img1_path <- "variants_af_files/figure-gfm/afcurve.png"
img1 <- readPNG(img1_path, native = TRUE, info = TRUE)
include_graphics(img1_path)

```

```{r, echo=FALSE, warning=FALSE}
af_list <- c('AF','EUR_AF' ,'PL_AF', 'gnomAD_FIN_AF', 'gnomAD_NFE_AF')
group.colors <- c(PL_AF = "#BC0020", AF = "#27384A", gnomAD_FIN_AF ="#48C095", gnomAD_NFE_AF = "#B6B6B6",EUR_AF='#5F5F5F')

```

```{r ACMG, echo=FALSE, warning=FALSE}
acmg <- read.table('../input/diseases/acmg_ready.tsv',sep='\t',header=T)

comp_acmg <- acmg %>%
   select(Uploaded_variation, all_of(af_list)) %>%
  mutate(across(AF:gnomAD_NFE_AF,as.character)) %>%
   mutate(across(AF:gnomAD_NFE_AF,as.numeric)) 

af <- comp_acmg %>% ggplot(aes(x=AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('1000 genomes AF') +
  ylab('Polish AF') +
  theme_classic()

eur <- comp_acmg %>% ggplot(aes(x=EUR_AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('1000 genomes European AF') +
  ylab('Polish AF') +
  theme_classic()

fin <- comp_acmg %>% ggplot(aes(x=gnomAD_FIN_AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('gnomAD Finnish AF') +
  ylab('Polish AF') +
  theme_classic()

nonfin <- comp_acmg %>% ggplot(aes(x=gnomAD_NFE_AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('gnomAD Non-Finnish European AF') +
  ylab('Polish AF') +
  theme_classic()



grid.arrange(af,eur,fin,nonfin,
             top=textGrob('ACMG genes'))


```


```{r clinvar, echo=FALSE, warning=FALSE}
clinsig <- read.table('../input/diseases/clin_sig_ready.tsv',sep='\t',header=T)
comp_clinsig <- clinsig %>% 
  select(Uploaded_variation, all_of(af_list)) %>%
  mutate(across(AF:gnomAD_NFE_AF,as.character)) %>%
  mutate(across(AF:gnomAD_NFE_AF,as.numeric)) 

af <- comp_clinsig %>% ggplot(aes(x=AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('1000 genomes AF') +
  ylab('Polish AF') +
  theme_classic()

eur <- comp_clinsig %>% ggplot(aes(x=EUR_AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('1000 genomes European AF') +
  ylab('Polish AF') +
  theme_classic()

fin <- comp_clinsig %>% ggplot(aes(x=gnomAD_FIN_AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('gnomAD Finnish AF') +
  ylab('Polish AF') +
  theme_classic()

nonfin <- comp_clinsig %>% ggplot(aes(x=gnomAD_NFE_AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('gnomAD Non-Finnish European AF') +
  ylab('Polish AF') +
  theme_classic()

library(gridExtra)

grid.arrange(af,eur,fin,nonfin,
             top=textGrob('ClINVAR pathogenic'))



```



```{r putative, echo=FALSE, warning=FALSE}
putative <- fread('../input/diseases/putative_ready.tsv',header=T)
comp_putative <- putative %>% 
  select(Uploaded_variation, all_of(af_list)) %>%
  mutate(across(AF:gnomAD_NFE_AF,as.character)) %>%
  mutate(across(AF:gnomAD_NFE_AF,as.numeric)) 

af <- comp_putative %>% ggplot(aes(x=AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('1000 genomes AF') +
  ylab('Polish AF') +
  theme_classic()

eur <- comp_putative %>% ggplot(aes(x=EUR_AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('1000 genomes European AF') +
  ylab('Polish AF') +
  theme_classic()

fin <- comp_putative %>% ggplot(aes(x=gnomAD_FIN_AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('gnomAD Finnish AF') +
  ylab('Polish AF') +
  theme_classic()

nonfin <- comp_putative %>% ggplot(aes(x=gnomAD_NFE_AF,y=PL_AF)) + 
  geom_point(col="#48C095") + 
  xlab('gnomAD Non-Finnish European AF') +
  ylab('Polish AF') +
  theme_classic()

library(gridExtra)

grid.arrange(af,eur,fin,nonfin,
             top = textGrob("Putative variants"))


```



```{r echo=FALSE, warning=FALSE}
library(png)
img1_path <- "variants_af_files/figure-gfm/impact_all_af.png"
img1 <- readPNG(img1_path, native = TRUE, info = TRUE)
include_graphics(img1_path)

```

## Number of variants in functional categories

```{r AF and IMPACT, echo=FALSE}
common <- read.table('../output/common_summary.tsv',header = T)
mediumrare <- read.table('../output/mediumrare_summary.tsv',header = T)
rare <- read.table('../output/rare_summary.tsv',header=T)

variants <- rbind(common,mediumrare,rare)
kable(variants %>% pivot_wider(names_from = IMPACT, values_from = n, values_fill = 0) %>%
        relocate(MODERATE, .before = MODIFIER)
        )

```







