---
title: "Structural variants.\newline Results for 1076 unrelated individuals"
subtitle:
output: 
  github_document: default
editor_options: 
  chunk_output_type: console
---

```{r setup, echo=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo=FALSE, dev = 'jpeg')
suppressMessages(library(knitr))
suppressMessages(library(tidyverse))
suppressMessages(library(data.table))

diff <- function(l) { 
  return(ifelse(length(l)==2, l[2]-l[1], NA))
}

#sv.af <- read.table('../input/sv_multisample_210519.smoove.square.unrelated.anno.HWE.tsv.gz', header=T)
#colnames(sv.af)[1:6] <- c('chrom','pos', 'svtype', 'len', 'ac', 'an')
#sv.af$chrom = factor(sv.af$chrom, levels = unique(sv.af$chrom))
#sv.af$af <- sv.af$ac/sv.af$an
#sv.af$len <- abs(as.numeric(sv.af$len))
#sv.af$CIPOS95 <- sapply(sv.af$CIPOS95, function(x) diff(as.numeric(strsplit(x,',')[[1]])))
#sv.af$CIEND95 <- sapply(sv.af$CIEND95, function(x) diff(as.numeric(strsplit(x,',')[[1]])))

sv.pers <- fread('../input/sv_multisample_210716.per_sample_stats.tsv', stringsAsFactors = T, na.strings = '.')
colnames(sv.pers) = c('sampleid','ID','chrom','AC','AN','svtype','svlen',
                      'HWE','MSHQ','SU','SR','PE','GT','GQ','DP','AB','DHFFC','SHQ')
sv.pers$af = abs(sv.pers$AC/sv.pers$AN)
sv.pers$svlen = abs(as.numeric(sv.pers$svlen))
if (all(levels(sv.pers$GT) == c('0/1','1/1'))) {
  levels(sv.pers$GT) <- c('het','hom')
} else if (all(levels(sv.pers$GT) == c('1/1','0/1'))) {
  levels(sv.pers$GT) <- c('hom','het')
} else {
  message(levels(sv.pers$GT))
}

annots = sv.pers %>% #select(ID, AC, AN, HWE, GQ, DP, AB, DHFFC) %>% 
  group_by(ID) %>% mutate(MGQ=mean(GQ), MDP=mean(DP), MAB=mean(AB), MDHFFC=mean(DHFFC))

# get unique records by chunk to not exhaust memory
sv.af = data.table()
chunk.size=1e6
MAX = dim(annots)[1]
for (i in 1:(MAX/chunk.size)) {
  cat(i)
  start = 1 + (chunk.size*(i-1))
  end   = min(chunk.size*i, MAX)
  sv.af=rbindlist(list(sv.af, annots[start:end,c(2:12,19:23)] %>% unique()))
}
  
sv.af = sv.af %>% unique()
sv.af$chrom = factor(sv.af$chrom, levels = unique(sv.af$chrom))

```

## QC

```{r set.filters, echo=T}

sv.af.full = sv.af
sv.af.full$PASS = (sv.af.full$MSHQ>=3 | sv.af.full$MSHQ==-1) & 
             (sv.af.full$svtype!='DEL' | sv.af.full$MDHFFC < 0.7) & 
             (sv.af.full$svtype!='DUP' | sv.af.full$MDHFFC > 1.25) &
             (sv.af.full$SR/sv.af.full$AC)>1 
             # & sv.af.full$HWE>0.1
           
sv.af = sv.af.full[sv.af.full$PASS,]
# save sv.af
# write.table(sv.af, row.names = F, quote = F, sep='\t',
#            file = '../input/sv_multisample_210716.smoove.square.unrelated.af_stats_PASS.tsv')

sv.pers.full = sv.pers
sv.pers.full$PASS= (sv.pers.full$MSHQ>=3 | sv.pers.full$MSHQ==-1) & 
                   (sv.pers.full$svtype!='DEL' | sv.pers.full$DHFFC < 0.7) & 
                   (sv.pers.full$svtype!='DUP' | sv.pers.full$DHFFC > 1.25) &
                   (sv.pers.full$SR/sv.pers.full$AC)>1
                   # & (sv.pers.full$HWE>0.1)

sv.pers = sv.pers.full[sv.pers.full$PASS,]

```

```{r plot.qc, echo=F, eval=F}

sv.af.full %>% 
  arrange(HWE) %>% 
  group_by(svtype,HWE) %>% 
  summarise(n=n()) %>% 
  mutate(cs=cumsum(n)) %>% 
  ggplot(aes(x=HWE, y=cs, color=svtype)) + geom_line() + ylab("Cummulative sum of variants") # + geom_vline(xintercept = 0.01)

# HWE per chromosome
sv.af.full %>% ggplot(aes(x=chrom, y=HWE)) + geom_boxplot()

# HWE-filter needs to omit chrX and Y 
sv.af.full %>% filter(PASS) %>% ggplot(aes(x=chrom, y=HWE)) + geom_boxplot()

sv.af.full %>% filter(svtype!='BND' & PASS) %>% 
  ggplot(aes(x=log10(svlen), y=HWE, color=svtype)) + geom_point() +
  ylab("HWE") + geom_vline(xintercept = 0.01)

kable(sv.af.full %>% filter(svtype!='BND' & PASS) %>% 
  group_by(svtype) %>% summarize(mlen=mean(svlen), medlen=median(svlen)))

kable(sv.pers %>% 
        group_by(svtype, sampleid) %>% 
        summarise(count=n(), count.pass=sum(PASS)) %>% mutate() %>%
        summarise(min=min(count), mean=mean(count), max=max(count), 
                  min.pass=min(count.pass), mean.pass=mean(count.pass), max.pass=max(count.pass))
      )

kable(sv.pers %>% filter(svtype != 'BND' & PASS) %>%
  group_by(svtype, sampleid) %>% 
  summarise(cumlen=sum(svlen)) %>%
  summarise(min=min(cumlen)/1e6, mean=mean(cumlen)/1e6, median=median(cumlen)/1e6, max=max(cumlen)/1e6) 
)

#sv.pers %>% filter(svtype!='BND' & PASS) %>% arrange(-svlen)


```

## Counts of structural variants (SV)

```{r sv.af.counts, echo=F}

kable(sv.af.full %>% 
        group_by(svtype) %>% 
        summarise(count=n(), count.PASS=sum(PASS), pct.pass=sum(PASS)/n()))

```


```{r sv.af.count.by.chromosomes, echo=F}

ggplot(sv.af, aes(x=chrom, fill=svtype)) +
         geom_bar(position = 'dodge') +
  xlab('Chromosome') +
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))

```

Fig. Figure presents counts of four SV types (deletions (DEL), duplications (DUP), inversion (INV), and break-ends (BND)) in autosomal and sex chromosomes.


```{r sv.individual.counts.table, echo=F}

kable(sv.pers %>% 
  group_by(svtype, sampleid) %>% 
  summarise(cnt=n()) %>% mutate() %>%
  summarise(min=min(cnt), mean=mean(cnt), median=median(cnt), max=max(cnt)) 
)

kable(sv.pers %>% 
  group_by(svtype, GT, sampleid) %>% 
  summarise(cnt=n()) %>% mutate() %>%
  summarise(min=min(cnt), mean=mean(cnt), median=median(cnt), max=max(cnt)) 
)
```

Table. Numbers of structural variant types per sample (A), with breakdown into homozygous and heterozygous (B)


```{r sv.individual.counts.plot, echo=F}

sv.pers %>% 
  group_by(sampleid, svtype) %>% 
  summarize(cnt=n()) %>% 
  ggplot(aes(x=svtype, y=cnt)) + geom_violin(fill='#48C095',col='#27384A') + geom_boxplot(width=0.1) +
  xlab('SV type') + ylab('Count')

```

Fig. Distribution of structural variant counts per sample, for the four SV categories.



## Lengths of structural variants (SV)


```{r sv.individual.lengths.table, echo=F}

kable(sv.pers %>% filter(svtype != 'BND') %>%
  group_by(svtype, sampleid) %>% 
  summarise(cumlen=sum(svlen)) %>%
  summarise(min=min(cumlen)/1e6, mean=mean(cumlen)/1e6, median=median(cumlen)/1e6, max=max(cumlen)/1e6) 
)


kable(sv.pers %>% filter(svtype != 'BND') %>%
  group_by(svtype, GT, sampleid) %>% 
  summarise(cumlen=sum(svlen)) %>%
  summarise(min=min(cumlen), mean=mean(cumlen), median=median(cumlen), max=max(cumlen)) 
)


```

Table. Total length of structural variant per sample, and per type (A), with breakdown into homozygous and heterozygous (B)


```{r sv.individual.lengths.plot, echo=F}

sv.pers %>% filter(svtype!='BND') %>%
  group_by(sampleid, svtype) %>% 
  summarize(len=sum(svlen)) %>% 
  ggplot(aes(x=svtype, y=len)) + geom_violin(fill='#48C095',col='#27384A') + geom_boxplot(width=0.1) + scale_y_log10() +
  xlab('SV type') + ylab('Total length per individual')

```



## Allele frequency of structural variants (SV)

```{r sv.af.cumsum, echo=F}

sv.af.cs  <-  sv.af %>% 
             arrange(af) %>%
             group_by(svtype, af) %>%
             summarise(cnt=n()) %>% 
             summarise(af, cs=cumsum(cnt), cspct=100*cumsum(cnt)/sum(cnt)) %>% 
             select(-cs)
  
group.colors <- c(BND = "#BC0020", DEL = "#27384A", DUP ="#48C095", INV = "blue")

sv.af.cs %>%
ggplot() + 
  geom_line(aes(x = af, y=cspct, color=svtype)) +
  scale_x_log10()+
  xlab('Allele frequency (log-scale)') + ylab('% variants') + 
  ylim(0,100) +
  scale_color_manual(values = group.colors,name='') +
  theme_minimal() +
  theme(legend.position = 'top')

```

Fig. Cumulative fraction of variants presenting with given allele frequency (on log-scale). Deletions (DEL), duplications (DUP), inversion (INV), and break-ends (BND).

### Allele frequency per individual

```{r sv.pers, echo=F, eval=F}

df <- sv.pers %>%
#    arrange(af) %>%
#    group_by(svtype, sampleid, af) %>%
#    summarise(cnt=n()) %>% 
#   mutate(af, cs=cumsum(cnt), cspct=100*cumsum(cnt)/sum(cnt)) %>% 
#    mutate(af, min=min(cspct), median=median(cspct), max=max(cspct)) %>% 
#   select(-cs)
  
#group.colors <- c(BND = "#BC0020", DEL = "#27384A", DUP ="#48C095", INV = "blue")

```


## SV Lengths

```{r sv.len.hist, echo=F}

sv.len <- sv.af %>% 
  select(svtype, svlen) %>% 
  filter(svtype!='BND')


ggplot(sv.len, aes(x=svlen, fill=svtype)) +
  geom_histogram(position = 'dodge', bins=80) +
  scale_x_log10() + 
  xlab('Length (log-scale)')

ggplot(sv.len, aes(x=svlen)) +
  geom_density(aes(color=svtype), stat='density', position='identity')  +
  scale_x_log10() + 
  xlab('Length (log-scale)')

ggplot(sv.len, aes(x=svlen, fill=svtype)) +
  geom_histogram(position = 'dodge', bins=80) + xlim(0,2000) + xlab('Length (<=2kb)')


```

Fig1. Distribution of SV lengths (log10 scale) for three SV types: deletions (DEL), duplications (DUP), and inversion (INV)

Fig2. Distribution of SV lengths (log10 scale) presented as density for the three SV types: deletions (DEL), duplications (DUP), and inversion (INV)

Fig3. Distribution of SV lengths in the range 0-1000bp, for the three SV types: deletions (DEL), duplications (DUP), and inversion (INV)


```{r echo=F}

del.cnt = (sv.len %>% filter(svtype == 'DEL') %>% select(svlen) %>% summarise(n()))
dup.cnt = (sv.len %>% filter(svtype == 'DUP') %>% select(svlen) %>% summarise(n()))
inv.cnt = (sv.len %>% filter(svtype == 'INV') %>% select(svlen) %>% summarise(n()))
```

Majority of deletions ware shortar than 1kb (`r 100 * dim(sv.len[sv.len$svtype=='DEL' & sv.len$svlen<1000,])[1] / del.cnt` %) and as much as `r 100 * dim(sv.len[sv.len$svtype=='DEL' & sv.len$svlen<1e4,])[1] / del.cnt`% were below 10kb

Only `r 100 * dim(sv.len[sv.len$svtype=='DEL' & sv.len$svlen>1e6,])[1] / del.cnt`% of deletions were longer than 1Mb.

Only `r 100 * dim(sv.len[sv.len$svtype=='DUP' & sv.len$svlen>1e6,])[1] / dup.cnt`% of duplications were longer than 1Mb.

As much as `r 100 * dim(sv.len[sv.len$svtype=='INV' & sv.len$svlen>1e6,])[1] / inv.cnt`% of inversions were longer than 1Mb.


### SV sizes

#### Deletions

```{r sv.lenstats.del, echo=F}
kable(summary(sv.len %>% filter(svtype=='DEL') %>% select(svlen)))
```

#### Duplications

```{r sv.lenstats.dup, echo=F}
kable(summary(sv.len %>% filter(svtype=='DUP') %>% select(svlen)))
```

#### Inversions

```{r sv.lenstats.inv, echo=F}
kable(summary(sv.len %>% filter(svtype=='INV') %>% select(svlen)))
```





